### PR TITLE
Run picotool step in github action

### PR DESCRIPTION
**Pull request by jannic**
This should catch bugs like https://github.com/rp-rs/rp-hal/issues/853

### CODE REVIEW

File: .github/workflows/rp2040_hal_examples.yml [hunk 1]
```diff
@@ -5,14 +5,18 @@ env:
...
+    runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@stable
         with:
           target: ${{ env.TARGET }}
       - name: Build
         run: cd ${PACKAGE} && cargo build
+      - name: Install picotool
+        run: sudo sh -c "apt-get install -y debian-archive-keyring && echo 'deb [signed-by=/usr/share/keyrings/debian-archive-keyring.gpg] http://deb.debian.org/debian testing main contrib non-free' >/etc/apt/sources.list.d/debian.list && apt update && apt-get install -y libxml2-utils picotool"
```
Comment (thejpster) [inline .github/workflows/rp2040_hal_examples.yml:17]: can this be a multi-line script? the one-liner is hard to read. Also the mix of apt-get and apt is weird.

```suggestion
        run:
          sudo apt-get update
          sudo apt-get install -y debian-archive-keyring libxml2-utils
          echo 'deb [signed-by=/usr/share/keyrings/debian-archive-keyring.gpg] http://deb.debian.org/debian testing main contrib non-free' | sudo tee /etc/apt/sources.list.d/debian.list
          sudo apt-get update
          sudo apt-get install -y picotool"
```

File: .github/workflows/rp2040_hal_examples.yml [hunk 2]
```diff
@@ -5,14 +5,18 @@ env:
...
+      - name: Test picotool
+        run: picotool info ${PACKAGE}/target/${TARGET}/debug/binary_info_demo -t elf -a | tee /dev/stderr | grep -q "rp2040-hal Binary Info Example"
```
Comment (thejpster) [inline .github/workflows/rp2040_hal_examples.yml:19]: What does tee'ing to /dev/stderr do here?

File: .github/workflows/rp235x_hal_examples_arm.yml [hunk 3]
```diff
@@ -5,14 +5,18 @@ env:
...
+    runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@stable
         with:
           target: ${{ env.TARGET }}
       - name: Build
         run: cd ${PACKAGE} && cargo build --target=${TARGET}
+      - name: Install picotool
+        run: sudo sh -c "apt-get install -y debian-archive-keyring && echo 'deb [signed-by=/usr/share/keyrings/debian-archive-keyring.gpg] http://deb.debian.org/debian testing main contrib non-free' >/etc/apt/sources.list.d/debian.list && apt update && apt-get install -y libxml2-utils picotool"
```
Comment (thejpster) [inline .github/workflows/rp235x_hal_examples_arm.yml:17]: Same comment as above.

File: .github/workflows/rp2040_hal_examples.yml (hunk 2 above)
Comment (jannic) [reply .github/workflows/rp2040_hal_examples.yml:19]: `grep` consumes stdout and checks for some expected string - that's what makes the test fail or pass. But then, I also want to see the output of `picotool`, if only to understand the error cause if `grep` doesn't find the expected string. So `tee /dev/stderr` copies the output of `picotool` to stderr where it won't be affected by the following `grep`.

File: .github/workflows/rp2040_hal_examples.yml (hunk 1 above)
Comment (jannic) [reply .github/workflows/rp2040_hal_examples.yml:17]: BTW if you know a better way to install a current version of picotool, I'm all ears. Installing a package from debian testing on ubuntu may be a bit unreliable.

Comment (thejpster) [reply .github/workflows/rp2040_hal_examples.yml]: https://github.com/raspberrypi/pico-sdk-tools/releases/tag/v2.0.0-5

Grab the tarball and unpack it?

**Review by thejpster — APPROVED** [review]

File: .github/workflows/rp2040_hal_examples.yml (hunk 1 above)
Comment (jannic) [reply .github/workflows/rp2040_hal_examples.yml]: Good suggestion, I didn't know that they provide binaries. https://github.com/raspberrypi/picotool doesn't mention it in the readme, and the releases page of that repo only has a source tarball.

As the pipeline works as it is, I'll merge this PR, but I consider simplifying the pipeline when I find some time to do it.

**Pull request merged**